music_assistant/providers/radioparadise/__init__.py

```diff
@@ -0,0 +1,369 @@
+                    provider_domain=self.domain,
+                    provider_instance=self.instance_id,
+                    item_id=channel_id,
+                    available=True,
+                )
+            },
+        )
+        radio.metadata.description = channel_info.get("description")
+
+        # Fetch the current metadata to get the latest cover image for the browse view.
+        metadata = await self._get_channel_metadata(channel_id)
+        cover_url = cast("str | None", metadata.get("cover_url")) if metadata else None
+
+        if cover_url:
+            images = [
+                MediaItemImage(
+                    provider=self.lookup_key,
+                    type=ImageType.THUMB,
+                    path=cover_url,
+                    remotely_accessible=True,
+                )
+            ]
+            radio.metadata.images = UniqueList(images)
+
+        return radio
+
+    async def _get_channel_metadata(self, channel_id: str) -> dict[str, Any] | None:
+        """Get current playing metadata for a channel."""
+        if channel_id not in self._channels_cache:
+            return None
+
+        channel_info = self._channels_cache[channel_id]
+        api_url = channel_info["api_url"]
+
+        try:
+            timeout = aiohttp.ClientTimeout(total=10)
+            async with (
+                aiohttp.ClientSession(timeout=timeout) as session,
+                session.get(api_url) as response,
+            ):
+                if response.status != 200:
+                    self.logger.debug(f"API call to {api_url} failed with status {response.status}")
+                    return None
+
+                data = await response.json()
+                if "artist" in data:
+                    current_song = data
+                elif "song" in data and len(data["song"]) > 0:
+                    current_song = data["song"][0]
+                else:
+                    self.logger.debug(f"No song data in API response for channel {channel_id}")
+                    return None
+
+                return {
+                    "title": current_song.get("title", ""),
+                    "artist": current_song.get("artist", ""),
+                    "cover_url": current_song.get("cover"),
+                    "duration": current_song.get("time"),
+                }
+        except Exception as exc:
+            self.logger.debug(f"Failed to get metadata for channel {channel_id}: {exc}")
+            return None
+
+    def _build_stream_url(self, channel_id: str) -> str:
+        """Build stream URL for a channel with fallback to other formats."""
+        if channel_id not in self._channels_cache:
+            return ""
+
+        channel_info = self._channels_cache[channel_id]
+        stream_urls = channel_info.get("stream_urls", {})
+
+        current_format_index = -1
+        with contextlib.suppress(ValueError):
+            current_format_index = FALLBACK_ORDER.index(self._stream_format)
```

Comment:
you will need to have a default value for current_format_index in case you run into the (ignored) ValueError so better use a try...except block here

Reply:
Code removed during rewrite